src/mapping/config/MappingConfiguration.cpp

```diff
@@ -160,15 +165,16 @@ void MappingConfiguration::xmlTagCallback(
 {
   PRECICE_TRACE(tag.getName());
   if (tag.getNamespace() == TAG) {
-    std::string   dir            = tag.getStringAttributeValue(ATTR_DIRECTION);
-    std::string   fromMesh       = tag.getStringAttributeValue(ATTR_FROM);
-    std::string   toMesh         = tag.getStringAttributeValue(ATTR_TO);
-    std::string   type           = tag.getName();
-    std::string   constraint     = tag.getStringAttributeValue(ATTR_CONSTRAINT);
-    Timing        timing         = getTiming(tag.getStringAttributeValue(ATTR_TIMING));
-    double        shapeParameter = 0.0;
-    double        supportRadius  = 0.0;
-    double        solverRtol     = 1e-9;
+    std::string   dir                       = tag.getStringAttributeValue(ATTR_DIRECTION);
+    std::string   fromMesh                  = tag.getStringAttributeValue(ATTR_FROM);
+    std::string   toMesh                    = tag.getStringAttributeValue(ATTR_TO);
+    std::string   type                      = tag.getName();
+    std::string   constraint                = tag.getStringAttributeValue(ATTR_CONSTRAINT);
+    Timing        timing                    = getTiming(tag.getStringAttributeValue(ATTR_TIMING));
+    double        shapeParameter            = 0.0;
+    double        supportRadius             = 0.0;
+    bool          gaussianUseShapeParameter = true;
```

Comment:
This will become easier to handle later on if you use a struct + enum to encode the RBF parameter;

```cpp
struct RBFParameter {

  enum struct Type {
    ShapeParameter,
    SupportRadius
  };

  Type type;
  double value;
}

// ... later in the RBF section

PREICICE_CHECK(isfinite(shapeParameter) ^ isfinite(supportRadius), "None set");
PREICICE_CHECK(isfinite(shapeParameter) && isfinite(supportRadius), "Both set");

RBFParameter param;
if (isfinite(shapeParameter)) {
  param.type = RBFParameter::Type::ShapeParameter;
  param.value = shapeParameter;
else {
  param.type = RBFParameter::Type::ShapeParameter;
  param.value = shapeParameter;
}

createMapping( ... param);

```

This will allow to simplify the createMapping in the future.

Reply:
Alright, I changed according to this approach.

Reply:
Looks like we had a misunderstanding.
The method I propose above encapsulates the kind (shape-parameter or support-radius) and the value into a single type.
This allows to replace both parameters `double shapeParameter` and `double supportRadius` by a single parameter of type `RBFParamter`. This then saves additional work in `createMapping`.

As it stands, you don't need the `struct RBFParamter` at all, you can simply replace it by the `enum struct Type`.
Also, you don't need the member `_rbfParameter` as you already deduce this information in the function where it is needed.

Reply:
So you propose to store this information for all rbf data mappings within this struct?

Reply:
I removed now the member variable `_rbfParameter` and the shape parameter as well as the support radius from the function arguments. They were now replaced by the `RBFParameter`. Is this now what you had in mind?

Reply:
Exactly! This monstrosity of a function looks a bit cleaner now. :tada: